parseAndPopulate/resolvers/expiration.py

```diff
@@ -0,0 +1,124 @@
+
+        Arguments:
+            :param module               (dict) Module with all the metadata
+            :param LOGGER               (logging.Logger) formated logger with the specified name
+            :param datatracker_failures (list) list of url that failed to get data from Datatracker
+            :param redis_connection     (RedisConnection) Connection used to communication with Redis
+        """
+        self.module = module
+        self.logger = logger
+        self.datatracker_failures = datatracker_failures
+        self.redis_connection = redis_connection
+
+    def resolve(self) -> t.Optional[bool]:
+        reference = self.module.get('reference')
+        expired = 'not-applicable'
+        expires = None
+        if self.module.get('maturity-level') == 'ratified':
+            expired = False
+            expires = None
+        if reference is not None and 'datatracker.ietf.org' in reference:
+            ref = reference.split('/')[-1]
+            rev = None
+            if ref.isdigit():
+                ref = reference.split('/')[-2]
+                rev = reference.split('/')[-1]
+            url = f'https://datatracker.ietf.org/api/v1/doc/document/?name={ref}&states__type=draft&states__slug__in=active,RFC&format=json'
+            retry = 6
+            while True:
+                try:
+                    response = requests.get(url)
+                    break
+                except Exception as e:
+                    retry -= 1
+                    self.logger.warning(
+                        f'Failed to fetch file content of {ref}')
+                    time.sleep(10)
+                    if retry == 0:
+                        self.logger.error(
+                            f'Failed to fetch file content of {ref} for 6 times in a row - SKIPPING.')
+                        self.logger.error(e)
+                        self.datatracker_failures.append(url)
+                        return None
+
+            if response.status_code == 200:
+                data = response.json()
+                objs = data.get('objects', [])
+                if len(objs) == 1:
+                    if rev == objs[0].get('rev'):
+                        rfc = objs[0].get('rfc')
+                        if rfc is None:
+                            expires = objs[0]['expires']
+                            expired = False
+                        else:
+                            expired = True
+                            expires = None
+                    else:
+                        expired = True
+                        expires = None
+                else:
+                    expired = True
+                    expires = None
+
+        expired_changed = self.__expired_change(
+            self.module.get('expired'), expired)
+        expires_changed = self.__expires_change(
+            self.module.get('expires'), expires)
+
+        if expires_changed or expired_changed:
```

Comment:
Please try to change some of these condition in this file to reduce number of indentation levels.
For example if you "reverse" this condition and return False, then whole code bellow can be reduced by one level of indentation.